core/modes/simulate_mode.go

```diff
@@ -5,25 +5,43 @@ import (
 
 	"github.com/SpectoLabs/hoverfly/core/matching"
 	"github.com/SpectoLabs/hoverfly/core/models"
+	"github.com/SpectoLabs/hoverfly/core/handlers/v2"
+	"strings"
 )
 
 type HoverflySimulate interface {
-	GetResponse(models.RequestDetails) (*models.ResponseDetails, *matching.MatchingError)
+	GetResponse(models.RequestDetails, bool) (*models.ResponseDetails, *matching.MatchingError)
 	ApplyMiddleware(models.RequestResponsePair) (models.RequestResponsePair, error)
 }
 
 type SimulateMode struct {
 	Hoverfly HoverflySimulate
+	MatchingStrategy string
 }
 
-func (this *SimulateMode) SetArguments(arguments ModeArguments) {}
+func (this *SimulateMode) View() v2.ModeView {
+	return v2.ModeView{
+		Mode: Simulate,
+		Arguments: v2.ModeArgumentsView{
+			MatchingStrategy: &this.MatchingStrategy,
+		},
+	}
+}
+
+func (this *SimulateMode) SetArguments(arguments ModeArguments) {
+	if arguments.MatchingStrategy == nil {
+		this.MatchingStrategy = "strongest"
+	} else {
+		this.MatchingStrategy = *arguments.MatchingStrategy
+	}
+}
 
 func (this SimulateMode) Process(request *http.Request, details models.RequestDetails) (*http.Response, error) {
 	pair := models.RequestResponsePair{
 		Request: details,
 	}
 
-	response, matchingErr := this.Hoverfly.GetResponse(details)
+	response, matchingErr := this.Hoverfly.GetResponse(details, strings.ToUpper(this.MatchingStrategy) != "FIRST")
```

Comment:
I'm not sure I like this. I think that maybe the decision as to which matching strategy is used should be made from within GetResponse, rather than forcing each caller of the `GetResponse()` function to have to determine it themselves. This is also a public function on the Hoverfly so it could potentially be in use by others.

Moving the decision into GetResponse() should also keep things simpler if we decide to adopt alternative matching strategies.